src/main/java/com/purbon/kafka/topology/TopicManager.java

```diff
@@ -67,15 +76,22 @@ public void sync(Topology topology) throws IOException {
       List<String> topicsToBeDeleted = new ArrayList<>();
       listOfTopics.stream()
           .forEach(
-              originalTopic -> {
-                if (!updatedListOfTopics.contains(originalTopic)) {
-                  topicsToBeDeleted.add(originalTopic);
+              topic -> {
+                if (!updatedListOfTopics.contains(topic) && !isAnInternalTopics(topic)) {
+                  topicsToBeDeleted.add(topic);
                 }
               });
       adminClient.deleteTopics(topicsToBeDeleted);
     }
   }
 
+  private boolean isAnInternalTopics(String topic) {
+    return internalTopicPrefixes.stream()
```

Comment:
What about `internalTopicPrefixes.stream().filter(topic::startsWith).count() > 0`?

Reply:
need to pass the prefix to startsWith, but I like it!